aws-xray-recorder-sdk-sql/src/main/java/com/amazonaws/xray/sql/TracingStatement.java

```diff
@@ -97,8 +97,24 @@ public Object invoke(Object proxy, Method method, Object[] args) throws Throwabl
 
             Subsegment subsegment = createSubsegment();
             if (subsegment == null) {
-                // don't trace if failed to create subsegment
-                return method.invoke(delegate, args);
+                try {
```

Comment:
This is quite a duplicated code. I'd recommend refactoring it to
```
           try {
                // execute the query 
                return method.invoke(delegate, args);
            } catch (Throwable t) {
                Throwable rootThrowable = t;
                if (t instanceof InvocationTargetException) {
                    // the reflection may wrap the actual error with an InvocationTargetException.
                    // we want to use the root cause to make the instrumentation seamless
                    InvocationTargetException ite = (InvocationTargetException) t;
                    if (ite.getTargetException() != null) {
                        rootThrowable = ite.getTargetException();
                    } else if (ite.getCause() != null) {
                        rootThrowable = ite.getCause();
                    }
                   ...
          
            if(subsegment != null) {
                 subsegment.addException(rootThrowable);
            }
            throw rootThrowable;
 ...
```. 
and merge with the code below